agent/agentcontainers/api.go

```diff
@@ -1146,12 +1146,30 @@ func (api *API) maybeInjectSubAgentIntoContainerLocked(ctx context.Context, dc c
 			codersdk.DisplayAppPortForward:    true,
 		}
 
+		// We need to read the config twice. This first time is so that we can extract the
+		// agent's name.
+		if config, err := api.dccli.ReadConfig(ctx, dc.WorkspaceFolder, dc.ConfigPath, []string{}); err != nil {
+			api.logger.Error(ctx, "unable to read devcontainer config", slog.Error(err))
+		} else {
+			// NOTE(DanielleMaywood):
+			// We only want to take an agent name specified in the root customization layer.
+			// This restricts the ability for a feature to specify the agent name. We may revisit
+			// this in the future, but for now we want to restrict this behavior.
+			if name := config.Configuration.Customizations.Coder.Name; name != "" {
```

Comment:
If the name didn't change, we could let this be the only invocation of `ReadConfig` couldn't we? But we'll need to pass the correct envs along here already to do that. We can defer the optimization for later though.